crac/src/main/java/io/micronaut/crac/netty/NettyEmbeddedServerCracHander.java

```diff
@@ -43,37 +41,38 @@ public class NettyEmbeddedServerCracHander implements OrderedResource {
 
     private static final Logger LOG = LoggerFactory.getLogger(NettyEmbeddedServerCracHander.class);
 
-    private final ApplicationEventPublisher<BeforeCheckpointEvent> beforeCheckpointEventPublisher;
-    private final ApplicationEventPublisher<AfterRestoreEvent> afterRestoreEventPublisher;
+    private final CracEventPublisher eventPublisher;
     private final NettyEmbeddedServer server;
 
     public NettyEmbeddedServerCracHander(
-        ApplicationEventPublisher<BeforeCheckpointEvent> beforeCheckpointEventPublisher,
-        ApplicationEventPublisher<AfterRestoreEvent> afterRestoreEventPublisher,
+        CracEventPublisher eventPublisher,
```

Comment:
We need to move this logic to its own class. Whether you are using Netty or not we should trigger a refresh event. 

I think we need a 

`@Singleton class RefreshEventResource implements OrderedResource` which does the refresh event trigger.

Reply:
Really good point.  I've pushed an ordered resource for this.  And renamed the `netty` package to `resources` as it's silly having a class per package